crates/miden-objects/src/batch/proven_batch.rs

```diff
@@ -20,13 +20,21 @@ pub struct ProvenBatch {
     input_notes: InputNotes<InputNoteCommitment>,
     output_notes: Vec<OutputNote>,
     batch_expiration_block_num: BlockNumber,
+    transactions: Vec<TransactionHeader>,
 }
 
 impl ProvenBatch {
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
 
     /// Creates a new [`ProvenBatch`] from the provided parts.
+    ///
+    /// Note that the transaction headers must be in the same order as the transactions in the
+    /// proposed batch. The order of the nullifiers and output note IDs within the header must also
+    /// match the order of input and output notes in the corresponding transaction. This is not
+    /// enforced by this type. The rationale for this requirement is that it allows a client to
+    /// cheaply validate the correctness of the transactions in a proven batch returned by a remote
+    /// prover.
```

Comment:
Should we maybe enforce _all_ this via new-types?

Something like `OrderedXXX(Vec<_>)` which can only be constructed from a `ProposedBatch`, or maybe an `BatchTransactions` (or whatever ordered set of txs abstraction we have/make).

Reply:
Yes, at least for the tx headers I think that makes sense. I added `OrderedBatches` and `OrderedTransactionHeaders` and will push those when I regain access.